### PR TITLE
feat: commandhandler now has functions to edit/get/delete responses

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR adds functions to `dpp::commandhandler` to allow users to edit, get, and delete, responses. Whilst `dpp::commandhandler` is deprecated, we still have a good few people using it.

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [ ] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: src/dpp/commandhandler.cpp [hunk 1]
```diff
@@ -421,20 +421,65 @@ void commandhandler::reply(const dpp::message &m, command_source source, command
...
 }
 
-void commandhandler::thonk(command_source source, command_completion_event_t callback)
+void commandhandler::thonk(command_source source, bool ephemeral, command_completion_event_t callback)
 {
-	thinking(source, callback);
+	thinking(source, ephemeral, callback);
+}
+
+void commandhandler::edit_response(const message &m, command_source source, command_completion_event_t callback) {
```
Comment (braindigitalis) [inline src/dpp/commandhandler.cpp:443]: these all need a check like in thinking, e.g.:

```cpp
if (!source.command_token.empty() && source.command_id) {
```

they cant do anything if its a message command.

File: src/dpp/commandhandler.cpp [hunk 2]
```diff
@@ -421,20 +421,65 @@ void commandhandler::reply(const dpp::message &m, command_source source, command
...
+
+void commandhandler::get_original_response(command_source source, command_completion_event_t callback) {
+	owner->post_rest(API_PATH "/webhooks", owner->me.id.str(), source.command_token + "/messages/@original", m_get, "", [creator = owner, cb = std::move(callback)](json& j, const http_request_completion_t& http) {
+		if (cb) {
+			cb(confirmation_callback_t(creator, message().fill_from_json(&j), http));
+		}
+	});
+}
+
+void commandhandler::edit_original_response(const message &m, command_source source, command_completion_event_t callback) {
```
Comment (braindigitalis) [inline src/dpp/commandhandler.cpp:459]: if we're adding these, shouldnt they also work for message commands?

this is the point of commandhandler, and why we dont really add much to it

**Review by braindigitalis — CHANGES_REQUESTED** [review]

**Comment by braindigitalis** [comment]
any movement on this?

**Comment by Jaskowicz1** [comment]
Closing, we are officially deprecating command handler so we should not support new features.

**Pull request closed without merging**